packages/core/core/src/InternalAsset.js

```diff
@@ -127,7 +127,9 @@ export default class InternalAsset {
     if (
       // $FlowFixMe
       typeof contentStream.bytesRead === 'number' &&
-      contentStream.bytesRead > 0
+      // If the amount of data read from this stream so far isn't exactly the amount
+      // of data that is available to be read, then it has been read from.
+      contentStream.bytesRead !== contentStream.readableLength
```

Comment:
The [readableEnded](https://nodejs.org/dist/latest-v12.x/docs/api/stream.html#stream_readable_readableended) property of all streams sounds perfect for this case... Unfortunately, it's only available in node 12.9+. 😢

Reply:
I think that can only tell is if the stream has ended, rather than one that may have been read from but not yet ended. The latter might be more useful for detecting problems.